extensions/common/vault/vault-aws/src/main/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVault.java

```diff
@@ -71,15 +72,21 @@ public AwsSecretsManagerVault(SecretsManagerClient smClient, Monitor monitor, Aw
     @Override
     public Result<Void> storeSecret(String key, String value) {
         var sanitizedKey = sanitizer.sanitizeKey(key);
-        var request = CreateSecretRequest.builder().name(sanitizedKey)
-                .secretString(value).build();
         try {
-            monitor.debug(String.format("Storing secret '%s' to AWS Secrets manager", sanitizedKey));
-            smClient.createSecret(request);
+            var updateSecretRequest = UpdateSecretRequest.builder().secretId(sanitizedKey).secretString(value).build();
+            smClient.updateSecret(updateSecretRequest);
+            monitor.debug(String.format("Secret '%s' updated in AWS Secrets Manager", sanitizedKey));
             return Result.success();
-        } catch (RuntimeException serviceException) {
-            monitor.severe(serviceException.getMessage(), serviceException);
-            return Result.failure(serviceException.getMessage());
+        } catch (ResourceNotFoundException e) {
```

Comment:
This way we're not catching other possible exceptions thrown by `updateSecret`, while we're catching them when calling `createSecret` (becasue catch at line 86).
For consistency, I would catch them as well

Reply:
It makes sense, it's worth adding a catch block to catch exceptions at runtime.